File: backend/pkg/models/workflow.go
package models

import (
	"time"
)

// Workflow represents the evolutionary definition of a logic pipeline.
type Workflow struct {
	ID           string                 `json:"id"`          // Unique Version ID
	TenantID     string                 `json:"tenant_id"`   // Multi-tenancy isolation
	WorkflowID   string                 `json:"workflow_id"` // Stable Concept ID
	Version      int                    `json:"version"`
	IsLatest     bool                   `json:"is_latest"`
	Name         string                 `json:"name"`
	Description  string                 `json:"description"`
	Status       string                 `json:"status"`
	ParentID     *string                `json:"parent_id,omitempty"`   // Hierarchy: points to ID (version) of parent
	ElementType  string                 `json:"element_type"`          // workflow, element, detail
	InputSchema  map[string]interface{} `json:"input_schema"`
	OutputSchema map[string]interface{} `json:"output_schema"`
	CreatedBy    string                 `json:"created_by"`
	CreatedAt    time.Time              `json:"created_at"`
	UpdatedAt    time.Time              `json:"updated_at"`

	// UI/API Control fields (not in DB table)
	SaveAsNewVersion bool `json:"save_as_new_version,omitempty"`
}

File: frontend/src/types/workflow.ts
export type WorkflowStatus = 'draft' | 'active' | 'archived';
export type ElementType = 'workflow' | 'element' | 'detail';

export interface Workflow {
  id: string;
  workflow_id: string;
  tenant_id: string;
  name: string;
  description: string;
  status: WorkflowStatus;
  version: number;
  is_latest: boolean;
  parent_id?: string;
  element_type: ElementType;
  input_schema: Record<string, any>;
  output_schema?: Record<string, any>;
  created_by: string;
  created_at: string;
  updated_at: string;
}

export interface WorkflowUpdatePayload extends Partial<Workflow> {
  save_as_new_version?: boolean;
}

export interface Tenant {
  id: string;
  name: string;
  domain: string;
  logo_svg?: string;
  brand_title?: string;
  created_at: string;
  updated_at: string;
}

export interface GroundingRule {
  id: string;
  tenant_id: string;
  workflow_id?: string;
  name: string;
  content: string;
  is_global: boolean;
  created_at: string;
  updated_at: string;
}

export interface HealthStatus {
  service: string;
  status: string;
  timestamp: string;
  version: string;
}
